### PR TITLE
[luajit] Add support for Linux and osx

### LINKED ISSUE #11809: [luajit] build failure
**Host Environment**
- OS: [macos 10.15.4]
- Compiler: revision

**To Reproduce**
`vcpkg install luajit --triplet x64-osx`

**Failure logs** 
```
Kais-MacBook-Pro:minetest tom$ vcpkg install luajit --triplet x64-osx
Computing installation plan...
The following packages will be built and installed:
    luajit[core]:x64-osx
Starting package 1/1: luajit:x64-osx
Building package luajit[core]:x64-osx...
-- Using cached /Users/tom/github-cpp/vcpkg/downloads/LuaJIT-LuaJIT-v2.0.5.tar.gz
-- Using source at /Users/tom/github-cpp/vcpkg/buildtrees/luajit/src/v2.0.5-045e1daa09
-- Building x64-osx-dbg
CMake Error at scripts/cmake/vcpkg_execute_required_process_repeat.cmake:23 (message):
    Command failed: /Users/tom/github-cpp/vcpkg/buildtrees/luajit/src/v2.0.5-045e1daa09/src/msvcbuild.bat /Users/tom/github-cpp/vcpkg/buildtrees/luajit/src/v2.0.5-045e1daa09/src dynamic debug static
    Working Directory: /Users/tom/github-cpp/vcpkg/buildtrees/luajit/x64-osx-dbg
    See logs for more information:
      /Users/tom/github-cpp/vcpkg/buildtrees/luajit\build-x64-osx-dbg-out.log
      /Users/tom/github-cpp/vcpkg/buildtrees/luajit\build-x64-osx-dbg-err.log

Call Stack (most recent call first):
  ports/luajit/portfile.cmake:25 (vcpkg_execute_required_process_repeat)
  scripts/ports.cmake:90 (include)


Error: Building package luajit:x64-osx failed with: BUILD_FAILED
Please ensure you're using the latest portfiles with `.\vcpkg update`, then
submit an issue at https://github.com/Microsoft/vcpkg/issues including:
  Package: luajit:x64-osx
  Vcpkg version: 2020.02.04-unknownhash

Additionally, attach any relevant sections from the log files above.
```
**Additional context**
Add any other context about the problem here, such as what you have already tried to resolve the issue.

### PR DESCRIPTION
**Pull request by NancyLi1013**
https://github.com/microsoft/vcpkg/blob/cb8a9fe7157f8ad2423742f893955ed9ced05910/ports/luajit/portfile.cmake#L27

Since `lujit `needs to build with the script` msvcbuild.bat` that requires Visual stuido.
So it only supports Windows platform.

Fix #11809

Note: No feature needs to test.

Edit: 
Use `make `to build on Linux and osx.

### CODE REVIEW

File: ports/luajit/CONTROL [hunk 1]
```diff
@@ -1,5 +1,5 @@
 Source: luajit
-Version: 2.0.5-3
+Version: 2.0.5-4
 Homepage: https://github.com/LuaJIT/LuaJIT
 Description: LuaJIT is a Just-In-Time (JIT) compiler for the Lua programming language.
-Supports: !uwp
\ No newline at end of file
+Supports: x86|x64|windows
```
Comment (JackBoosY) [inline ports/luajit/CONTROL:5]:
```suggestion
Supports: windows&(!uwp)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

**Review by JackBoosY — APPROVED** [review]

**Comment by NancyLi1013** [comment]
When building on Linux via `Makefile`, post the error:
```
CMake Error at scripts/cmake/vcpkg_build_make.cmake:250 (file):
  file RENAME failed to rename

    /home/vruil/vcpkg/packages/luajit_x64-linux_tmp/home/vruil/vcpkg/installed/x64-linux/

  to

    /home/vruil/vcpkg/packages/luajit_x64-linux

  because: No such file or directory
```
So I changed `file(RENAME "${CURRENT_PACKAGES_DIR}_tmp${_VCPKG_INSTALL_PREFIX}/" "${CURRENT_PACKAGES_DIR}")` as 
`file(RENAME "${CURRENT_PACKAGES_DIR}" "${CURRENT_PACKAGES_DIR}")` in `vcpkg_build_make.cmake`. In this case, I can build luajit without any errors on Linux.

@Neumann-A 
Could you please help take a look about this rename failure?

**Comment by JackBoosY** [comment]
Temporary close this PR, Please reopen it when continue this work.

**Pull request closed without merging**